### PR TITLE
#170764576 User can sign up

### PR DESCRIPTION
**Pull request by bihire**
### What does this Pr do ?
A new User can signup new credentials in the API
### Description of the task to be completed ?
- save new credentials  saved in a postgres database
- return error if the credentials exists already
- run unit tests on the endpoint
- create user table if they don't exist already
- hash the saved password
- return error if inputs don't meet validation standards
- return token success

have the following endpoint work 
- POST /auth/signup
### How should this be manually tested?
after cloning this repository
- Using postman test the endpoint above

key: Content-type value: application/json
- generate a jwt token to put in headers as "token" by creating a new user

then send the following info
- {	
    "firstName": "",
    "lastName": "",
    "email": "",
    "password": "",
    "userName":"",
    "role": ""
  }
### Any background context you want to provide?
N/A
### What are the relevant pivot tracker stories ?
#170764576
https://www.pivotaltracker.com/story/show/170764576

### Any screenshots?
image
image

### CODE REVIEW

**Review by niyodusengaclement — CHANGES_REQUESTED** [review]
This error should be handled first

**Review by niyodusengaclement — APPROVED** [review]
LGTM

File: src/models/user.js [hunk 1]
```diff
@@ -1,41 +1,41 @@
-export default (sequelize, DataTypes) => {
-  const User = sequelize.define('User', {
-    email: {
-      type: DataTypes.STRING,
-      allowNull: {
-        args: false,
-        msg: 'Please enter your email address'
+module.exports = (sequelize, DataTypes) => {
```
Comment (Elvis-rugamba) [inline src/models/user.js]: Could you use ES6?

**Comment by istafrica-dennw** [comment]
@bihire  this coverage is not acceptable. it is below the line unless there is a good exlpaination
 the branch of 50 %, etc is very tiny. we agreed to have at least 80% coverage.

**Review by ajimae — APPROVED** [review]
LGTM!

**Pull request merged**